delfin/drivers/driver.py

```diff
@@ -178,3 +178,104 @@ def list_shares(self, context):
+        status: Health status(normal, offline, abnormal, unknown)
+        ip_address: Ip address of the host
+        storage_id: Storage id at delfin side
+        """
+        raise NotImplementedError(
+            "Driver API list_storage_hosts() is not Implemented")
+
+    def list_storage_host_groups(self, context):
+        """List all storage host groups from storage system."""
+        """
+        *********Model description**********
+        native_storage_host_group_id: Native id of host grp at backend side
+                                      (mandatory)
+        name: Name of the host grp
+        description: Description of the host grp
+        storage_hosts: List of associated hosts if any(, separated list)
+        storage_id: Storage id at delfin side
+        """
+        raise NotImplementedError(
+            "Driver API list_storage_host_groups() is not Implemented")
+
+    def list_port_groups(self, context):
+        """List all port groups from storage system."""
+        """
+        *********Model description**********
+        native_port_group_id: Native id of port grp at backend side (mandatory)
+        name: Name of the port grp
+        description: Description of the port grp
+        ports: List of associated ports if any(, separated list)
+        storage_id: Storage id at delfin side
+        """
+        raise NotImplementedError(
+            "Driver API list_port_groups() is not Implemented")
+
+    def list_volume_groups(self, context):
+        """List all volume groups from storage system."""
+        """
+        *********Model description**********
+        native_volume_group_id: Native id of volume grp at backend side
+                                (mandatory)
+        name: Name of the volume grp
+        description: Description of the volume grp
+        volumes: List of associated volumes if any(, separated list)
+        storage_id: Storage id at delfin side
+        """
+        raise NotImplementedError(
+            "Driver API list_volume_groups() is not Implemented")
+
+    def list_masking_views(self, context):
+        """List all masking views from storage system."""
+        """
+        *********Model description**********
+        native_masking_view_id: Native id of volume grp at backend side
+                                (mandatory)
+        name: Name of the masking view
+        description: Description of the masking view
+        native_storage_host_group_id: Native id of host grp at backend side
+        native_port_group_id: Native id of port grp at backend side
+        native_volume_group_id: Native id of volume grp at backend side
+        native_storage_host_id: Native id of host at backend side
+        volumes: List of associated volumes(, separated list)
+        ports: List of associated ports, separated list)
+        storage_host_initiators: List of associated initiators
+                                 (, separated list)
```

Comment:
confused about the {
        volumes: List of associated volumes(, separated list)
        ports: List of associated ports, separated list)
        storage_host_initiators: List of associated initiators
                                 (, separated list)}

Reply:
Updated masking view description details